api/bots_api/bot_lib.py

```diff
@@ -51,7 +51,7 @@ def show_error_and_exit(self):
         sys.exit(1)
 
 def send_reply(message, response, email, send_message):
-    # type: (Dict[str, Any], str, str, Any) -> Dict[str, Any]
+    # type: (Dict[str, Any], str, str, Callable[[Dict[str, Any]], Dict[str, Any]]) -> Dict[str, Any]
```

Comment:
I've not run mypy myself at this time due to local development limitations, but I could see that it either is or should be possible, since at least the inferred types can be exposed via reveal_type(). That said, if `Any` is marked, mypy considers it as an explicit mark of 'dynamic' code, so presumably wouldn't pay attention in a normal validation run.

This is more of a general question than specifically related to this code; so perhaps ask on [#general](https://chat.zulip.org/#narrow/stream/general)?